### PR TITLE
MAINTAINERS.yml updates

### PR DESCRIPTION
**Pull request by nashif**
The changes here are the result of running all open pull-requests through a script to identify the assignee and finding many gaps or old data. Some of the changes here are coming from pending PRs that were waiting for some TSC confirmation, others are done based on pending maintainer and collaborator changes that were not submitting yet.

Two noteable changes:

- CMake/Build system: This is now limited to just the infrastructure and not every single Cmake file in the tree.
- Documentation: ditto, this is just for the doc infrastructure and not for every RST file in the tree.

The reason for the above two changes is the fact that most PRs would have some cmake or doc changes and assignees in this case should not be the owners of the build system or the doc infra, instead we should assign to the subsystem being changed. (Otherwise majority of PRs would be assigned to the doc and build system maintainers).

### CODE REVIEW

**Review by galak — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Review by erwango — APPROVED** [review]

File: MAINTAINERS.yml [hunk 1]
```diff
@@ -1093,7 +1149,92 @@ SPARC arch:
...
+    collaborators:
+        - jhedberg
+        - aasthagr
+    files:
+        - boards/x86/
+        - soc/x86/
+    labels:
+        - "platform: X86"
+
+Intel Platformas (Xtensa):
```
Comment (ioannisg) [inline MAINTAINERS.yml:1179]: Platforms*

**Review by ioannisg — CHANGES_REQUESTED** [review]
Just a typo.
Otherwise, this looks like a good enhancement.

**Review by ioannisg — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

File: MAINTAINERS.yml [hunk 2]
```diff
@@ -439,7 +461,13 @@ Documentation:
...
 
 "Drivers: ESPI":
-    status: orphaned
+    status: maintained
+    maintainers:
+      - albertofloyd
+    collaborators:
+      - VenkatKotakonda
+      - scottwcpg
+      - albertofloyd
```
Comment (henrikbrixandersen) [inline MAINTAINERS.yml:470]: This lists @albertofloyd as both maintainer and collaborator?

File: MAINTAINERS.yml [hunk 3]
```diff
@@ -1093,7 +1149,92 @@ SPARC arch:
...
+        - "platform: Intel CAVS"
+
+NXP Platforms:
+  status: maintained
+  maintainers:
+    - MaureenHelm
+  files:
+    - boards/arm/mimx*/
+    - boards/arm/frdm_k*/
+    - boards/arm/lpcxpress*/
```
Comment (henrikbrixandersen) [inline MAINTAINERS.yml:1200]: We could add `boards/arm/twr_*` here as well.

File: MAINTAINERS.yml (hunk 2 above)
Comment (nashif) [reply MAINTAINERS.yml:470]: no harm I guess, a maintainer is still a collab

File: MAINTAINERS.yml [hunk 4]
```diff
@@ -1095,7 +1173,93 @@ SPARC arch:
...
+    - drivers/*/*mchp*
+  labels:
+    - "platform: Microchip"
+
+nRF Platforms:
+    status: maintained
+    maintainers:
+        - ioannisg
+    files:
+        - boards/arm/nrf*/
```
Comment (mbolivar-nordic) [inline MAINTAINERS.yml:1254]: I think `boards/arm/*nrf*/` would be better, to catch things like `thingy52_nrf52832`.

File: doc/contribute/project_roles.rst [hunk 5]
```diff
@@ -194,7 +194,17 @@ Generic guidelines for deciding and filling in the Maintainers' list
 
   * Re-assigned by original assignee (see “Assignee” slide)
 
-* Updates to the MAINTAINERS file should be in a standalone PRs
+* In general, updates to the MAINTAINERS file should be
+  in a standalone commit alongside other changes introducing new files and
+  directories to the tree.
+* Major changes to the file, including the addition of new areas with new maintainers
+  should come in as standalone pull-requests.
+* Depending on the change, review by the TSC will be required. If additinal
```
Comment (mbolivar-nordic) [inline doc/contribute/project_roles.rst:202]: `additional`

File: doc/contribute/project_roles.rst [hunk 6]
```diff
@@ -196,7 +194,17 @@ Generic guidelines for deciding and filling in the Maintainers' list
 
   * Re-assigned by original assignee (see “Assignee” slide)
 
-* Updates to the MAINTAINERS file should be in a standalone PRs
+* In general, updates to the MAINTAINERS file should be
+  in a standalone commit alongside other changes introducing new files and
+  directories to the tree.
+* Major changes to the file, including the addition of new areas with new maintainers
+  should come in as standalone pull-requests and require TSC review.
```
Comment (ioannisg) [inline doc/contribute/project_roles.rst:201]: changes in maintainership

**Review by ioannisg — APPROVED** [review]

**Pull request merged**